src/main/java/frc/robot/subsystems/TurretSubsystem.java

```diff
@@ -95,31 +135,69 @@ public TurretSubsystem(RobotStateManager robotStateManager) {
               false,
               0);
       turretMech = new Mechanism2d(4, 4);
-      root = turretMech.getRoot("Root", 2, 2);
+      turretRoot = turretMech.getRoot("Root", 2, 2);
       turretAngleSim =
-          root.append(new MechanismLigament2d("Turret", 2, 0, 5, new Color8Bit(Color.kRed)));
+          turretRoot.append(new MechanismLigament2d("Turret", 2, 0, 5, new Color8Bit(Color.kRed)));
       turretTab.add("Turret", turretMech);
+
+      // pitch
+      pitchSim =
+          new SingleJointedArmSim(
+              DCMotor.getNEO(1),
+              Constants.TurretConstants.PITCH_CONVERSION_FACTOR,
+              Constants.TurretConstants.SHOOTER_MASS
+                  * Constants.TurretConstants.SHOOTER_CENTER_OF_GRAVITY
+                  * Constants.TurretConstants.SHOOTER_CENTER_OF_GRAVITY
+                  / 3,
+              Constants.TurretConstants.SHOOTER_CENTER_OF_GRAVITY,
+              Math.toRadians(Constants.TurretConstants.PITCH_MIN_ANGLE_DEGREES),
+              Math.toRadians(Constants.TurretConstants.PITCH_MAX_ANGLE_DEGREES),
+              true,
+              0);
+      pitchMech = new Mechanism2d(4, 4);
+      pitchRoot = pitchMech.getRoot("Root", 2, 2);
+      pitchAngleSim =
+          pitchRoot.append(new MechanismLigament2d("Pitch", 2, 0, 5, new Color8Bit(Color.kBlue)));
+      turretTab.add("Pitch", pitchMech);
     }
 
     turretMotor.restoreFactoryDefaults();
-    turretMotor.setSmartCurrentLimit(40);
+    turretMotor.setSmartCurrentLimit(Constants.TurretConstants.TURRET_SMART_CURRENT_LIMIT);
+
+    pitchMotor.restoreFactoryDefaults();
+    pitchMotor.setSmartCurrentLimit(Constants.TurretConstants.PITCH_SMART_CURRENT_LIMIT);
 
     this.robotStateManager = robotStateManager;
 
+    // Soft Limits
     turretMotor.setSoftLimit(
         CANSparkMax.SoftLimitDirection.kReverse,
         (float)
-            (-Constants.TurretConstants.MAX_TURRET_ANGLE_DEGREES
+            (Constants.TurretConstants.TURRET_MIN_ANGLE_DEGREES
```

Comment:
This might be more readable if you extract these calculations to their own variables, then pass those variables to the method call. E.g.:

```java
final double minAngleInRadiansMaybeIDK = Constants.TurretConstants.TURRET_MIN_ANGLE_DEGREES / (360 * Constants.TurretContants.TURRET_MOTOR_TURRET_RATIO);
...
turretMotor.setSoftLimit(Constants.TurretConstants.TURRET_MIN_ANGLE_DEGREES, minAngleInRadiansMaybeIDK);
```

Reply:
done